Fix pbs_hook_set_jobenv.py when mom is on remotehost

shilpakodli opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature
The tests of "pbs_hook_set_jobenv.py" are currently written in such a way that it will work when pbs_mom is present in on the same host where pbs_server is present.
For example in one of the test(test_interactive_no_hook) where you are submitting an interactive job(this will be launched on compute node but on service node) and in that shell we need to capture output of env variable in a file. Later on we are creating a key value pair from the output of "env" variable. This is done using "read_env" function. These tests are failing with keyError because, read_env is trying to read a non existing file.

#### Describe Your Change
-> Added a parameter "host" in log_lines where user can send hostname on which logs has to be searched.
-> Changed common_log_match function to search log files on corresponding hosts.
-> Copying files from mom host to server in read_env function


#### Link to Design Doc
<!--- If there is a design, link to it here: **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)** -->

[TestPbsHookSetJobEnv_logs.txt](https://github.com/openpbs/openpbs/files/5073702/TestPbsHookSetJobEnv_logs.txt)
[TestPbsHookSetJobEnv_logs_before_fix.txt](https://github.com/openpbs/openpbs/files/5073704/TestPbsHookSetJobEnv_logs_before_fix.txt)

Klovely commented on test/tests/functional/pbs_hook_set_jobenv.py:
```diff
@@ -127,6 +128,8 @@ def tearDown(self):
             os.remove(self.job_out2_tempfile)
             os.remove(self.job_out3_tempfile)
+            self.du.rm(self.mom.hostname, self.job_out1_tempfile)
+            self.du.rm(self.mom.hostname, self.job_out2_tempfile)
```
Notice ,Line 131 and 132 is throwing error in tearDown as below in test output logs:-
2020-08-14 03:13:47,142 INFOCLI2 r1i0n0: ssh r1i0n0.ib0.smc-default.chf.rdlabs.hpecorp.net /usr/bin/rm /tmp/job_out1rml3t0l8
2020-08-14 03:13:47,423 ERROR    err: ["/usr/bin/rm: cannot remove '/tmp/job_out1rml3t0l8': No such file or directory"]
2020-08-14 03:13:47,424 INFOCLI2 r1i0n0: ssh r1i0n0.ib0.smc-default.chf.rdlabs.hpecorp.net /usr/bin/rm /tmp/job_out2iv75u2ft
2020-08-14 03:13:47,701 ERROR    err: ["/usr/bin/rm: cannot remove '/tmp/job_out2iv75u2ft': No such file or directory"]

Think this is because we already remove file in line 128 and 129.

shilpakodli replied:
Hi Lovely, this error is ignorable as the file is not there, there is no harm to run an rm command on that.

shilpakodli commented:
Hi @Klovely , I have addressed your review comments. Attaching latest execution logs.
[TestPbsHookSetJobEnv_latest_logs.txt](https://github.com/openpbs/openpbs/files/5083930/TestPbsHookSetJobEnv_latest_logs.txt)

shilpakodli commented:
Attaching results where server and mom are on same host:
Description: Tests from given sources on platforms SUSE15, CENTOS8
Platforms: SUSE15,CENTOS8
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|3203|16|0|0|0|2|14|

Klovely commented:
Thanks @shilpakodli latest changes LGTM

bayucan commented:
Merged #1971 to master.
